sdk/python/kfp/containers/_build_image_api.py

```diff
@@ -0,0 +1,148 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the speci
+
+__all__ = [
+    'build_image_from_env',
+]
+
+
+import logging
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+import requests
+
+from . import get_default_image_builder
+from ..compiler._container_builder import ContainerBuilder
+
+
+def get_python_image() -> str:
+    return 'python' + ':{}.{}.{}'.format(sys.version_info[0], sys.version_info[1], sys.version_info[2])
+
+
+default_base_image = get_python_image
+
+
+_default_package_url_template = 'https://pypi.org/pypi/{}/json/'
+
+
+_package_info_cache = {}
+
+
+def _python_package_exists(name: str, version : str = None) -> str:
+    package_subdir = name + ('/' + version if version else '')
+    package_url = _default_package_url_template.format(package_subdir)
+    if package_url in _package_info_cache:
+        return True
+    
+    response = requests.get(package_url)
+    if response.ok:
+        _package_info_cache[package_url] = response.json()
+        return True
+
+    return False
+
+
+def generate_requirements_file(requirements_path: str):
+    logging.info('Generating the requirements.txt file')
+    import pkg_resources
+    with open(requirements_path, 'w') as f:
+        for dist in pkg_resources.working_set:
+            req = dist.as_requirement()
+            logging.info('Checking package: {}.'.format(req))
+
+            # Checking the existense of the particular package version and not just the package project as some system-installed packages exist on PyPI, but the version is old.
+            # ERROR: Could not find a version that satisfies the requirement python-apt===1.8.3- (from -r requirements.txt (line 195)) (from versions: 0.0.0, 0.7.8)
+            # ERROR: No matching distribution found for python-apt===1.8.3- (from -r requirements.txt (line 195))                        
+            if _python_package_exists(dist.project_name, dist.version):
+                f.write(str(req) + '\n')
+            else:
+                if _python_package_exists(dist.project_name, version=None):
+                    logging.warning('Package "{}" exists on PyPI, but version "{}" does not exist.'.format(dist.project_name, dist.version))
+                else:
+                    logging.warning('Package version "{}" does not exist on PyPI.'.format(req))
+
+
+_container_work_dir = '/python_env'
+
+
+def generate_dockerfile_text(context_dir: str, dockerfile_path: str):
+    # Generating the Dockerfile
+    logging.info('Generating the Dockerfile')
+
+    requirements_rel_path = 'requirements.txt'
+    requirements_path = os.path.join(context_dir, requirements_rel_path)
+    requirements_file_exists = os.path.exists(requirements_path)
+
+    # Creating requirements.txt from the list of installed packages if requirements.txt does not already exist.
```

Comment:
Generating requirements.txt will result in inconsistent behavior when user open the same working directory from different env. I'd suggest not adding it. 

It's very easy for user to do it by their own by command "!pip freeze > requirements.txt".

Reply:
Done.
The idea was that the reproducibility is achieved by other users using the image built by this function as opposed to other people building their own versions of the images.